mlir/lib/Dialect/Arith/IR/InferIntRangeInterfaceImpls.cpp

```diff
@@ -42,6 +42,12 @@ void arith::ConstantOp::inferResultRanges(ArrayRef<ConstantIntRanges> argRanges,
   }
   if (auto arrayCstAttr =
           llvm::dyn_cast_or_null<DenseIntElementsAttr>(getValue())) {
+    if (arrayCstAttr.isSplat()) {
+      setResultRange(getResult(), ConstantIntRanges::constant(
+                                      arrayCstAttr.getSplatValue<APInt>()));
```

Comment:
The alternative is that you can still cast to SplatElementsAttr and then check the element type. Either way is fine.